scripts/i2c_lcd_driver.py

```diff
@@ -0,0 +1,179 @@
+# -*- coding: utf-8 -*-
+# Original code found at:
+# https://gist.github.com/DenisFromHR/cc863375a6e19dce359d
+
+"""
+Compiled, mashed and generally mutilated 2014-2015 by Denis Pleic
+Made available under GNU GENERAL PUBLIC LICENSE
```

Comment:
We need t check, if the GPL license has any implications.

Reply:
I refert to [FAQ about the GNU Licenses](https://www.gnu.org/licenses/gpl-faq.en.html#DoesFreeSoftwareMeanUsingTheGPL).

Under [WhyUseGPL?](https://www.gnu.org/licenses/gpl-faq.en.html#WhyUseGPL) is stated that 

> Using the GNU GPL will require that all the released improved versions be free software.

First, it states that GPL is a free software licence. Second even modifications are allowed but we even did not this, we just use the lcd_driver code in a non-modified version. 

Last but not least we distribute the source code openly which is requested by GPL according to [GPLRequireSourcePostedPublic](https://www.gnu.org/licenses/gpl-faq.en.html#GPLRequireSourcePostedPublic). 

> if you release the modified version to the public

and again, we did not modify the code, we just use the code (the lcd_driver) here.

In conclusion, it is fine. 

Best regards,
Simon

Reply:
**Disclaimer: I’m not a lawyer** 

Phoniebox code is licensed under MIT license, which is a permissive  open source license. That means for example that you can relicense the code with very few restrictions. So you could integrate MIT licensed code in proprietary code.

In contrast GPL is a copyleft  open source license, which expects that all code distributed together grants same degrees of freedom, i.e. not used in proprietary code.

So from my point of view we cannot integrate GPL code in the Phoniebox code base.
I think we need opinions from @MiczFlor here of course.
One possibility would be to ask the original author of the code, if this code could be relicensed under a permissive open source license.

Reply:
Hi @s-martin @SimonChelkowski 
I fear that @s-martin is right, actually, well, I know that this is the case.
Which is a real shame... We can add this to the wiki. And refer to the wiki via the components folder. Should I do that?

Reply:
That's the only solution I'm afraid. We could ask the author of that code, too.

Reply:
Hi @SimonChelkowski @s-martin 
I made this wiki page:
https://github.com/MiczFlor/RPi-Jukebox-RFID/wiki/LED_displays_HD44780-i2c

Reply:
I will contact the author and ask for permission. In the meantime the wiki page is a very good idea and compromise. Thanks @MiczFlor for creating the wiki page.

Reply:
I contacted the author and ask him for help in resolving this issue [link](https://gist.github.com/DenisFromHR/cc863375a6e19dce359d#gistcomment-3221873). Let's wait and see how it develops.

Reply:
Hi @SimonChelkowski
I don't really mind if you or anybody else uses my code in any way - except in purely commercial projects.
So, as long as my code is used in Open Source projects (non-commercial), I'm OK with it.
Since RPi-Jukebox-RFID is free software (non-commercial), you have my explicit permission for using my 'RPi_I2C_driver.py' code.
Rgds,
Denis Pleic (DenisFromHR)

Reply:
@DenisFromHR thanks for the permission!

@MiczFlor, I guess we could merge it, when the other comments are resolved.